shapeshift/web
# perf: optimized asset search

## Description

This PR aims to improve rendering performance of the asset search modal, and makes minor changes to how assets are selected such that compute is repeated less often during component mount and rendering.

## Pull Request Type


## Issue (if applicable)



NA

## Risk



Moderate risk of regressions for the following:
- asset sorting in asset selection modals, wallet, anywhere that assets are sorted
- asset selection modal behavior generally
- trade input, namely asset selection

## Testing



The changes here are primarly around relacation of existing code and deletion of unused logic. This includes a fair bit of rewiring of some of the key logic relating to asset selection and asset searching, so its difficult to say exactly what should be tested, but as a general guide of where to focus testing efforts:

- check asset selection and trades are working the same way as production:
  - on the trade page
  - on the dashboard
  - on asset page
  - on fiat page
  - on arkeo page
  - on fox-ops page
- check asset lists throughout the app are being sorted in the same way as production
- check asset search is working in asset selection

### Engineering



See above.

### Operations



No change should be apparent aside from improved rendering performance, so regression testing is required.

These changes are not feature flagged, so will be immediately user facing on deploy to production. Please see testing instructions above for guide on the type of regression testing that should be completed before being approved for approved for deployment to production.

## Screenshots (if applicable)
